### ISSUE TITLE
Empty Map with nested Map Types raises an IndexError.

### ISSUE THREAD

**Issue by CaptainCuddleCube** [ClickHouse/clickhouse-connect#239]
### Describe the bug

If your query returns an empty map, but the map is of a next Map type, an IndexError is raised. 
The query works fine on the server.

### Steps to reproduce
1. `client.query("select Cast(([],[]), 'Map(String, Map(String, String))')")`

### Expected behaviour
It should not raise an error, and return an empty dictionary.

### Code example
```python
import clickhouse_connect
dsn='http://admin:dev@localhost:8123/default'
client = clickhouse_connect.get_client(dsn=dsn)
client.query("select Cast(([],[]), 'Map(String, Map(String, String))')")
# IndexError: array index out of range
```

#### Environment
* clickhouse-connect version: 0.6.11
* Python version: 3.11.5
* Operating system: Linux/MacOS

#### ClickHouse server
* ClickHouse Server version: 23.6.2.18
* ClickHouse Server non-default settings, if any: N/A
* `CREATE TABLE` statements for tables involved: N/A

**Comment by CaptainCuddleCube** [comment]
It's a one line fix in the source code, just update to the following line of code:
in file `clickhouse_connect/datatypes/containt.py:183`
```
        total_rows = 0 if len(offset) == 0 else offsets[-1]
```

**Comment by genzgd** [comment]
Fixed in 0.6.12
